moveit_ros/planning_interface/CMakeLists.txt

```diff
@@ -8,88 +8,89 @@ if(NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
-  moveit_ros_manipulation
-  moveit_ros_move_group
-  geometry_msgs
-  tf2
-  tf2_eigen
-  tf2_geometry_msgs
-  tf2_ros
-  roscpp
-  actionlib
-  rospy
-  rosconsole
-)
+find_package(ament_cmake REQUIRED)
+find_package(moveit_msgs REQUIRED)
+find_package(moveit_ros_planning REQUIRED)
+#find_package(moveit_ros_warehouse REQUIRED)
+#find_package(moveit_ros_manipulation REQUIRED)
+#find_package(moveit_ros_move_group REQUIRED)
+find_package(geometry_msgs REQUIRED)
+find_package(tf2 REQUIRED)
+find_package(tf2_eigen REQUIRED)
+find_package(tf2_geometry_msgs REQUIRED)
+find_package(tf2_ros REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(rclcpp_action REQUIRED)
 
-find_package(PythonInterp REQUIRED)
-find_package(PythonLibs "${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}" REQUIRED)
+#find_package(PythonInterp REQUIRED)
+#find_package(PythonLibs "${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}" REQUIRED)
 
 find_package(Boost REQUIRED)
-if(Boost_VERSION LESS 106700)
-  set(BOOST_PYTHON_COMPONENT python)
-elseif()
-  set(BOOST_PYTHON_COMPONENT python${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR})
-endif()
+#if(Boost_VERSION LESS 106700)
+#  set(BOOST_PYTHON_COMPONENT python)
+#elseif()
+#  set(BOOST_PYTHON_COMPONENT python${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR})
+#endif()
 
 find_package(Boost REQUIRED COMPONENTS
   date_time
   filesystem
   program_options
-  ${BOOST_PYTHON_COMPONENT}
+  #${BOOST_PYTHON_COMPONENT}
   system
   thread
 )
 find_package(Eigen3 REQUIRED)
-find_package(eigenpy REQUIRED)
+#find_package(eigenpy REQUIRED)
 
 set(THIS_PACKAGE_INCLUDE_DIRS
-  py_bindings_tools/include
-  common_planning_interface_objects/include
-  planning_scene_interface/include
-  move_group_interface/include
+  #py_bindings_tools/include
+  #common_planning_interface_objects/include
+  #planning_scene_interface/include
+  #move_group_interface/include
   moveit_cpp/include
 )
 
-catkin_python_setup()
-
-catkin_package(
-  LIBRARIES
-    moveit_common_planning_interface_objects
-    moveit_planning_scene_interface
-    moveit_move_group_interface
-    moveit_cpp
-  INCLUDE_DIRS
-    ${THIS_PACKAGE_INCLUDE_DIRS}
-  CATKIN_DEPENDS
-    actionlib
-    moveit_ros_planning
-    moveit_ros_warehouse
-    moveit_ros_manipulation
-    moveit_ros_move_group
-    geometry_msgs
-    moveit_msgs
-    tf2_geometry_msgs
-  DEPENDS
-    EIGEN3
+set(THIS_PACKAGE_LIBRARIES
+  #moveit_common_planning_interface_objects
+  #moveit_planning_scene_interface
+  #moveit_move_group_interface
+  moveit_cpp
 )
 
+
+set(THIS_PACKAGE_INCLUDE_DEPENDS
+  rclcpp_action
+  moveit_ros_planning
+  #moveit_ros_warehouse
+  #moveit_ros_manipulation
+  #moveit_ros_move_group
+  geometry_msgs
+  moveit_msgs
+  tf2_geometry_msgs
+)
+
+#catkin_python_setup()
+
 include_directories(${THIS_PACKAGE_INCLUDE_DIRS}
-                    ${Boost_INCLUDE_DIRS}
-                    ${catkin_INCLUDE_DIRS})
+                    ${rclcpp_INCLUDE_DIRS}
+                    ${Boost_INCLUDE_DIRS})
```

Comment:
Shouldn't these be included automatically when using `ament_target_dependencies` .?

Reply:
+1